src/collector/alloc.rs

```diff
@@ -121,6 +135,15 @@ impl GcAllocation {
                 // So we can run `finalize` here, right before deallocation
                 (&mut *(finalize_ptr as *mut dyn Finalize)).finalize();
             }
+            DeallocationAction::BoxDrop => {
+                // Safe as long as only boxed values are created with BoxDrop deallocate action
+                // Additionally, this is the only instance where the pointer should be alive so
+                // we are not taking it mutably anywhere else. The death of a pointer in action,
+                // really makes you think...
+                let box_ptr = Box::from_raw(scan_ptr as *mut dyn Scan);
```

Comment:
Is this correct? Don't we need to use exactly the result of `into_raw`?

Reply:
For clarification, are you saying that it should be `*mut T` instead?

Reply:
@Others ping, not sure if you saw the above

Reply:
@alekratz I looked into this and couldn't find a definitive answer. Pretty sure it's equivalent to coercing a box then dropping it, so pretty sure it's okay. Not blocking this PR

(MIRI doesn't complain either)